nixos/less configure less with module

JohnAZoidberg opened the pull request:
###### Motivation for this change
Because less is configured not directly with a plaintext file but the config file requires translation to a binary file I thought this is the perfect use case for a nixos module.
Not only can less be configured declaratively now, but I added the convenience that less by default now extracts (xz, gzip, bz2) compressed files transparently before reading.

I think everything possible in the real configuration can be done with the nix language now - therefore there is no need for an option like "extraConfig".

###### Things done

Added a module to configure less with. Extracts compressed files by default before opening them.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

JohnAZoidberg committed:
nixos/less configure less with module

JohnAZoidberg commented on nixos/modules/programs/less.nix:
```diff
@@ -0,0 +1,123 @@
+            "${pkgs.less}/bin/lesskey -o $out $src";
+
+  lessPipe = pkgs.writeScriptBin "lesspipe.sh" ''
+    #! /bin/sh
```
Should I change it to `#!${stdenv.shell}`?

Mic92 replied:
yes

Mic92 replied:
I wonder if [lesspipe](https://github.com/wofr06/lesspipe) would be the better default here. We already package it in nixpkgs.

JohnAZoidberg replied:
Yup, that's a good idea. It's (almost) always better to use something well maintained than to hack a script together yourself.

JohnAZoidberg committed:
nixos/less: use lesspipe package for preprocessing

Rather than a custom script the less config now uses the lesspipe
package config by default.

Mic92 commented:
Thanks. Module looks good to me with good examples.
